test/jasmine/tests/lib_test.js

```diff
@@ -2459,6 +2459,58 @@ describe('Test lib.js:', function() {
             expect(toContainer).toEqual(expected);
         });
     });
+
+    describe('concat', function() {
+        var concat = Lib.concat;
+        it('works with multiple Arrays', function() {
+            expect(concat([1], [[2], 3], [{a: 4}, 5, 6]))
+                .toEqual([1, [2], 3, {a: 4}, 5, 6]);
```

Comment:
`Lib.concat` is fantastic :tada: 

Could we `spyOn(Array.prototype, 'concat')` in this test-case to make sure it is called in the "all regular arrays" case?

Reply:
Good call - https://github.com/plotly/plotly.js/pull/3211/commits/1ca72c39344c54f086bf7d5f728337322c6d68fe
`toEqual` calls `Array.concat` profusely, so I had to rearrange a little to get this to work right...